src/bin/pg_autoctl/pgsql.c

```diff
@@ -1344,3 +1344,74 @@ pgsql_listen(PGSQL *pgsql, char *channels[])
 
 	return true;
 }
+
+
+/*
+ * pgsql_alter_extension_update_to executes ALTER EXTENSION ... UPDATE TO ...
+ */
+bool
+pgsql_alter_extension_update_to(PGSQL *pgsql,
+								const char *extname, const char *version)
+{
+	char command[BUFSIZE];
+	char *escapedIdentifier, *escapedVersion;
+	PGconn *connection = NULL;
+	PGresult *result = NULL;
+
+	/* open a connection upfront since it is needed by PQescape functions */
+	connection = pgsql_open_connection(pgsql);
+	if (connection == NULL)
+	{
+		/* error message was logged in pgsql_open_connection */
+		return false;
+	}
+
+	/* escape the extname */
+	escapedIdentifier = PQescapeIdentifier(connection, extname, strlen(extname));
+	if (escapedIdentifier == NULL)
+	{
+		log_error("Failed to update extension \"%s\": %s", extname,
+				  PQerrorMessage(connection));
+		pgsql_finish(pgsql);
+		return false;
+	}
+
+	/* escape the version */
+	escapedVersion = PQescapeIdentifier(connection, version, strlen(version));
+	if (escapedIdentifier == NULL)
+	{
+		log_error("Failed to update extension \"%s\" to version \"%s\": %s",
+				  extname, version,
+				  PQerrorMessage(connection));
+		pgsql_finish(pgsql);
+		return false;
+	}
+
+	/* now build the SQL command */
+	snprintf(command, BUFSIZE, "ALTER EXTENSION %s UPDATE TO %s",
```

Comment:
guess we want to error when `snprintf` needed to cutoff the string formatting due to reaching `BUFSIZE`. What woud the behaviour be in the current implementation? Guess it sends an incomplete query to postgres :(

Reply:
Fixed.

Reply:
Oh and 1024 bytes for `ALTER EXTENSION %s UPDATE TO %s`, and with the extension name of type `name` (limited to 64 bytes including the null byte at the end of the string), you need a pretty long version string before that can happen. I'd happily call it a bug. That said, we can implement some checking, that's easy and might save us in case of strange data corruption on file names or something.